persistence/src/main/java/de/cyface/persistence/DatabaseMigrator.kt

```diff
@@ -0,0 +1,757 @@
+                                Constants.TAG,
+                                "Not enough geoLocations to update distance in measurement entry:$measurementId"
+                            )
+                            continue
+                        }
+
+                        // Calculate distance for selected measurement
+                        var distance = 0.0
+                        var lastLocationLat: Double? = null
+                        var lastLocationLon: Double? = null
+                        while (geoLocationCursor.moveToNext()) {
+                            val latColumnIndex = geoLocationCursor.getColumnIndex("lat")
+                            val lonColumnIndex = geoLocationCursor.getColumnIndex("lon")
+                            val newLocationLat =
+                                geoLocationCursor.getFloat(latColumnIndex).toDouble()
+                            val newLocationLon =
+                                geoLocationCursor.getFloat(lonColumnIndex).toDouble()
+
+                            // We cannot calculate a distance from just one geoLocation:
+                            if (lastLocationLat == null || lastLocationLon == null) {
+                                @Suppress("KotlinConstantConditions")
+                                Validate.isTrue(lastLocationLat == null && lastLocationLon == null)
+                                lastLocationLat = newLocationLat
+                                lastLocationLon = newLocationLon
+                                continue
+                            }
+
+                            // Calculate distance between last two locations
+                            val newDistance = calculateDistance(
+                                lastLocationLat,
+                                lastLocationLon,
+                                newLocationLat,
+                                newLocationLon
+                            )
+                            Validate.isTrue(newDistance >= 0)
+                            distance += newDistance
+                            lastLocationLat = newLocationLat
+                            lastLocationLon = newLocationLon
+                        }
+                        Log.v(
+                            Constants.TAG,
+                            "Updating distance for measurement $measurementId to $distance"
+                        )
+                        database.execSQL("UPDATE measurements SET distance = $distance WHERE _id = $measurementId")
+                    }
+                } finally {
+                    measurementCursor?.close()
+                    geoLocationCursor?.close()
+                }
+            }
+
+            /**
+             * Calculates the distance between two locations like in the [DefaultDistanceCalculation]
+             * implementation at the time when version `10` was introduced.
+             */
+            private fun calculateDistance(
```

Comment:
While I migrated the migration code, I noticed that in some places I didn't use hard-coded stuff, e.g. I used `DefaultDistanceCalculation` which might change in the future, but the migration should stay static, so I added the distance calculation algorithm used at the point of that migration step here, so it does not change.